api/v1alpha1/bgppeer_webhook.go

```diff
@@ -41,6 +42,20 @@ func (bgpPeer *BGPPeer) SetupWebhookWithManager(mgr ctrl.Manager) error {
 
 //+kubebuilder:webhook:verbs=create;update,path=/validate-metallb-io-v1alpha1-bgppeer,mutating=false,failurePolicy=fail,groups=metallb.io,resources=bgppeers,versions=v1alpha1,name=bgppeervalidationwebhook.metallb.io,sideEffects=None,admissionReviewVersions=v1
 
+var _ webhook.Defaulter = &BGPPeer{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (bgpPeer *BGPPeer) Default() {
+	bgppeerlog.Info("default", "name", bgpPeer.Name)
+
+	if bgpPeer.Spec.HoldTime == 0 {
+		bgpPeer.Spec.HoldTime = 90 * time.Second
+	}
+	if bgpPeer.Spec.KeepaliveTime == 0 {
+		bgpPeer.Spec.KeepaliveTime = 30 * time.Second
```

Comment:
Here we are missing the HT/3 default behaviour.

Reply:
how ?

Reply:
In metallb, if HT is set but KA is not, aren't we setting KA to HT/3 ? I assumed that since we are redoing the defaulting here, we keep the same logic.

Reply:
right but we don't need to that here since metallb will do that, the above are default if the field is not set plus what we are worried about here from validation is if KA is configure more than HT hence I don't think we need to duplicate this here

Reply:
If a user specifies HT but not KA, KA will get 30 seconds regardless of HT and the validation will fail, while the configuration is legit from metallb side.

Reply:
That's what I suggested in https://github.com/metallb/metallb-operator/pull/106#discussion_r749252613 , but your counterargument seemed convincing :-)

I am fine with both.

Reply:
If we don't default ka, I think it's fair not to default heartbeat either.

Reply:
u mean holdtime ? if we don't default it and some configure keepalive to large value we will be hossed :)

Reply:
dropped all default for now its not working for some reason